tools/i18n/unescape-contents

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+import argparse
+import html
+import json
+import os
+import sys
+from contextlib import contextmanager
+from pathlib import PurePath
+from typing import IO, Any, Generator, NamedTuple, Union
+
+
+class CLIArgs(NamedTuple):
+    filename: str
+    indent_level: int
+
+
+def parse_args() -> CLIArgs:
+    parser = argparse.ArgumentParser(
+        prog="unescape-contents",
+        description="Unescape Unicode and HTML entities in a JSON file",
+    )
+
+    parser.add_argument("filename", type=str)
+    parser.add_argument("--indent-level", dest="indent_level", type=int, default=2, required=False)
+    args = parser.parse_args()
+
+    return CLIArgs(args.filename, args.indent_level)
+
+
+@contextmanager
+def tempfile_for_atomic_overwrite(
+    destination: Union[str, os.PathLike[str]]
+) -> Generator[IO[Any], None, None]:
+    """
+    Creates a sibling temporary file in the same directory as `destination` and
+    yields the file-like object returned by `open()`. Upon completion of the
+    context block, the file handle is closed and the temporary file is removed,
+    either by replacing `destination` with its contents (see `os.replace()`
+    documentation) if the context succeeded, or by `os.unlink()`-ing it if the
+    context failed.
+
+    This is not a general-purpose temporary file creator: for usecases that
+    don't require moving the temporary file instead of deleting it, use the
+    `tempfile` module.
+    """
+
+    source_path = PurePath(os.path.abspath(destination))
+    temp_path = source_path.with_name(f".{source_path.name}.unescaped")
+    temp_file = open(temp_path, "w+")
+
+    try:
+        yield temp_file
+        temp_file.close()
+        os.replace(temp_file.name, destination)
+    finally:
+        try:
+            temp_file.close()
+            os.unlink(temp_path)
+        except FileNotFoundError:
+            # In the event of success, we've already moved this file
+            pass
```

Comment:
I implemented this as a `finally` mostly to avoid re-raise noise if this were implemented as `except Exception as e`, despite the fact that this inner `try` will *only* do anything useful in the event of a raised Exception. I'm in no way tied to doing it this way, so open to any subjective feedback on this that comes up.

Reply:
I think the temporary file logic probably unnecessary complexity for a tool we expect to use and run into errors running as infrequently as this one. I'd probably have not written the try/except/finally here.

Reply:
Yeah, personally what I'd have written would be to just overwrite the file in place. The script isn't particularly likely to crash, and if it does, you can reset the state of the target file easily via `git checkout HEAD <path>`, so I think the temporary file process just serves to obscure the main part of what this script is doing.

Reply:
Removed, that rationale seems good.